### PR TITLE
#145: Mock class not assert arguments with the empty arg list

### PR DESCRIPTION
**Pull request by yogyrton**
https://github.com/RonasIT/laravel-helpers/issues/145

### CODE REVIEW

File: src/Traits/MockTrait.php [hunk 1]
```diff
@@ -133,17 +129,24 @@ protected function assertArguments(
...
+        if ($expectedCount !== $actualCount) {
+            $optionalParametersCount = count(array_filter($actual, fn ($item) => $item === 'optionalParameter'));
+            $requiredParametersCount = $actualCount - $optionalParametersCount;
+
+            if ($expectedCount > $actualCount || $expectedCount < $requiredParametersCount) {
+                throw new Exception("Failed assert that function {$function} was called with {$expectedCount} arguments, actually it calls with {$actualCount} arguments.");
+            }
         }
 
+        $expected = array_merge($expected, array_fill(0, $actualCount - $expectedCount, 'optionalParameter'));
```
Comment (aizlee) [inline src/Traits/MockTrait.php:144]:
```suggestion
        $expected = array_pad($expected, $actualCount, 'optionalParameter');
```

File: src/Traits/MockTrait.php [hunk 2]
```diff
@@ -133,17 +129,24 @@ protected function assertArguments(
...
-            : "Namespace '{$class}'\nFunction: '{$function}'\nCall index: {$callIndex}";
-
         $expectedCount = count($expected);
         $actualCount = count($actual);
 
-        if ($expectedCount != $actualCount) {
-            throw new Exception("Failed assert that function {$function} was called with {$expectedCount} arguments, actually it calls with {$actualCount} arguments.");
+        if ($expectedCount !== $actualCount) {
+            $optionalParametersCount = count(array_filter($actual, fn ($item) => $item === 'optionalParameter'));
+            $requiredParametersCount = $actualCount - $optionalParametersCount;
```
Comment (aizlee) [inline src/Traits/MockTrait.php:137]:
```suggestion
            $requiredParametersCount = count(array_filter($actual, fn ($item) => $item !== 'optionalParameter'));
```

Comment (aizlee) [reply src/Traits/MockTrait.php:137]: @yogyrton  Could we use `optionalParameter` as constant of class?

**Review by aizlee — CHANGES_REQUESTED** [review]

File: src/Traits/MockTrait.php [hunk 3]
```diff
@@ -13,6 +13,8 @@ trait MockTrait
 {
     use PHPMock;
 
+    protected const OPTIONAL_PARAMETER = 'optionalParameter';
```
Comment (aizlee) [inline src/Traits/MockTrait.php:16]:
```suggestion
    protected const OPTIONAL_ARGUMENT_NAME = 'optionalParameter';
```

**Review by aizlee — CHANGES_REQUESTED** [review]

**Review by DenTray — APPROVED** [review]

**Review by aizlee — APPROVED** [review]

**Pull request merged**